lib/src/util/bolt_helpers.cc

```diff
@@ -188,6 +188,11 @@ namespace Util {
     fs::permissions(dir, NIX_DIR_PERMS);
   }
 
+  void createSymLink(const fs::path& destination, const fs::path& source) {
+    fs::create_symlink(destination, source);
+    fs::permissions(destination, NIX_DIR_PERMS);
```

Comment:
This is a race. In theory someone could access the symlink with default permissions in between creation and the permissions fix. We should be able to specify the permissions at creation time (on POSIX with UMASK, on Windows just as part of the CreateFile IIRC). Dunno how much of a refactor there is to achieve that...

Reply:
So after some digging on this:

It looks like there are ways for POSIX nodes to specify umask of the pxp-agent service, which means any file/symlink/dir created with boost should match the umask of the service process. That means between the file being created and the call to fs::permissions it looks like the user can still control the security of the files/symlinks/dirs.

For windows: all permissions are inherited by the containing directory, meaning there is no scenario where files/symlinks will have perms more open than the containing directories.

Since download_file isn't supporting specialized permissions from the user I believe the combination of those things means we don't need to worry about the race.

Reply:
Can you leave a comment near this code with a summary of that info for later?